src/api.js

```diff
@@ -131,14 +132,14 @@ var schema = function(validator, options) {
       // If we're supposed to validate outgoing messages and output schema is
       // defined, then we have to validate against it...
       if(options.output !== undefined && !options.skipOutputValidation) {
-        var errors = validator.check(json, options.output);
-        if (errors) {
+        var error = validate(json, options.output);
+        if (error) {
           debug("Reply for %s didn't match schema: %s got errors: %j from output: %j",
-                req.url, options.output, errors, json);
+                req.url, options.output, error, json);
           let err = new Error('Output schema validation of ' + options.output);
```

Comment:
I propose we do: ` new Error('Output schema validation error: ' + error)` as the message is pretty featureful right?